libyul/optimiser/Suite.cpp

```diff
@@ -87,6 +87,41 @@ using namespace std;
 using namespace solidity;
 using namespace solidity::yul;
 
+namespace
+{
+
+#ifdef OUTPUT_PERFORMANCE_METRICS
+void outputPerformanceMetrics(map<string, int64_t> const& _metrics)
+{
+	vector<pair<string, int64_t>> durations(_metrics.begin(), _metrics.end());
+	sort(
+		durations.begin(),
+		durations.end(),
+		[](pair<string, int64_t> const& _lhs, pair<string, int64_t> const& _rhs) -> bool
+		{
+			return _lhs.second < _rhs.second;
+		}
+	);
+	int64_t total = 0;
+	for (auto&& [step, count]: durations)
+		total += count;
+
+	cerr << "Performance metrics of optimizer steps" << endl;
+	cerr << "======================================" << endl;
+	for (auto&& [step, count]: durations)
+	{
+		double percentage = 100.0 * static_cast<double>(count) / static_cast<double>(total);
+		double sec = static_cast<double>(count) / 1000000.0;
+		cerr << setw(7) << setprecision(3) << percentage << "% (" << sec << " s): " << step << endl;
```

Comment:
This is just horrible. Why is there no easy way to align floats on the decimal point?

Reply:
```c++
fmt::format("{:>7.3f}% ({}) s: {}", percentage, sec, step);
```
Note that I haven't actually run this, so it might need tweaking - but yes, please use `fmt::format` for anything more verbose than adding two strings to output streams.